Black 21.5b1 fails to build on Darwin

**Describe the bug**

Fails during tests on unstable channel:

```
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform darwin -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/black-21.5b1, configfile: pyproject.toml
collected 211 items / 3 deselected / 208 selected

tests/test_black.py ...................................................F [ 25%]
F..............s......................x.x.x........                      [ 49%]
tests/test_blackd.py .............                                       [ 55%]
tests/test_format.py ................................................... [ 80%]
..................................                                       [ 96%]
tests/test_primer.py .......                                             [100%]

=================================== FAILURES ===================================
_________ BlackTestCase.test_output_locking_when_writeback_color_diff __________

self = <tests.test_black.BlackTestCase testMethod=test_output_locking_when_writeback_color_diff>

    @event_loop()
    def test_output_locking_when_writeback_color_diff(self) -> None:
        with cache_dir() as workspace:
            for tag in range(0, 4):
                src = (workspace / f"test{tag}.py").resolve()
                with src.open("w") as fobj:
                    fobj.write("print('hello')")
            with patch("black.Manager", wraps=multiprocessing.Manager) as mgr:
>               self.invokeBlack(["--diff", "--color", str(workspace)], exit_code=0)

tests/test_black.py:1169:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_black.py:139: in invokeBlack
    self.assertEqual(
E   AssertionError: 1 != 0 : Failed with args: ['--verbose', '--config', '/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/black-21.5b1/tests/empty.toml', '--diff', '--color', '/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/tmpov34r68o']
E   stdout: ''
E   stderr: '\nAborted!\n'
E   exception: 1
----------------------------- Captured stderr call -----------------------------
Process SyncManager-26:
Traceback (most recent call last):
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/managers.py", line 608, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/managers.py", line 154, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/connection.py", line 448, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/connection.py", line 591, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
____________ BlackTestCase.test_output_locking_when_writeback_diff _____________

self = <tests.test_black.BlackTestCase testMethod=test_output_locking_when_writeback_diff>

    @event_loop()
    def test_output_locking_when_writeback_diff(self) -> None:
        with cache_dir() as workspace:
            for tag in range(0, 4):
                src = (workspace / f"test{tag}.py").resolve()
                with src.open("w") as fobj:
                    fobj.write("print('hello')")
            with patch("black.Manager", wraps=multiprocessing.Manager) as mgr:
>               self.invokeBlack(["--diff", str(workspace)], exit_code=0)

tests/test_black.py:1156:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_black.py:139: in invokeBlack
    self.assertEqual(
E   AssertionError: 1 != 0 : Failed with args: ['--verbose', '--config', '/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/black-21.5b1/tests/empty.toml', '--diff', '/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/tmp448w4_oe']
E   stdout: ''
E   stderr: '\nAborted!\n'
E   exception: 1
----------------------------- Captured stderr call -----------------------------
Process SyncManager-27:
Traceback (most recent call last):
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/managers.py", line 608, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/managers.py", line 154, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/connection.py", line 448, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/multiprocessing/connection.py", line 591, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
=============================== warnings summary ===============================
tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
  <unknown>:107: DeprecationWarning: invalid escape sequence \

tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
  <unknown>:111: DeprecationWarning: invalid escape sequence \

tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
tests/test_black.py::BlackTestCase::test_docstring_no_string_normalization
  <unknown>:117: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:163: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:167: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:173: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:160: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:164: DeprecationWarning: invalid escape sequence \

tests/test_format.py::TestSimpleFormat::test_simple_format_15_docstring
  <unknown>:170: DeprecationWarning: invalid escape sequence \

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_black.py::BlackTestCase::test_output_locking_when_writeback_color_diff
FAILED tests/test_black.py::BlackTestCase::test_output_locking_when_writeback_diff
= 2 failed, 202 passed, 1 skipped, 3 deselected, 3 xfailed, 12 warnings in 23.49s =
builder for '/nix/store/qzqkn5s75q7njl7bxv61zhsm1kgrkws1-python3.8-black-21.5b1.drv' failed with exit code 1
error: build of '/nix/store/qzqkn5s75q7njl7bxv61zhsm1kgrkws1-python3.8-black-21.5b1.drv' failed
```

Seems to be caused by https://github.com/NixOS/nixpkgs/pull/120684.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A python38Packages.black`

**Expected behavior**

Build succeeds.

**Notify maintainers**

@sveitser 

**Metadata**

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.4.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.6`
 - channels(abutler): `"home-manager, nixpkgs-21.11pre294471.51bb9f3e9ab"`
 - nixpkgs: `/Users/abutler/.nix-defexpr/channels/nixpkgs`
 ```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - python38Packages.black
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/black/default.nix
+++ b/pkgs/development/python-modules/black/default.nix
@@ -5,6 +5,7 @@
 , appdirs
 , attrs
 , click
+, colorama
 , dataclasses
 , mypy-extensions
 , pathspec
@@ -12,17 +13,20 @@
 , regex
 , toml
 , typed-ast
-, typing-extensions }:
+, typing-extensions
+, uvloop
+}:
+
 
 buildPythonPackage rec {
   pname = "black";
-  version = "21.5b1";
+  version = "21.6b0";
 
   disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1cdkrl5vw26iy7s23v2zpr39m6g5xsgxhfhagzzflgfbvdc56s93";
+    sha256 = "016f6bhnnnbcrrh3cvmpk77ww0nykv5n1qvgf8b3044dm14264yw";
   };
 
   nativeBuildInputs = [ setuptools-scm ];
@@ -31,7 +35,7 @@ buildPythonPackage rec {
   # Black starts a local server and needs to bind a local address.
   __darwinAllowLocalNetworking = true;
 
-  checkInputs =  [ pytestCheckHook parameterized ];
+  checkInputs = [ pytestCheckHook parameterized ];
 
   preCheck = ''
     export PATH="$PATH:$out/bin"
@@ -39,6 +43,9 @@ buildPythonPackage rec {
     # The top directory /build matches black's DEFAULT_EXCLUDE regex.
     # Make /build the project root for black tests to avoid excluding files.
     touch ../.git
+  '' + lib.optionalString stdenv.isDarwin ''
+    # Work around https://github.com/psf/black/issues/2105
+    export TMPDIR="/tmp"
   '';
 
   disabledTests = [
@@ -55,19 +62,21 @@ buildPythonPackage rec {
     appdirs
     attrs
     click
+    colorama
     mypy-extensions
     pathspec
     regex
     toml
-    typed-ast
-    typing-extensions
-  ] ++ lib.optional (pythonOlder "3.7") dataclasses;
+    typed-ast # required for tests and python2 extra
+    uvloop
+  ] ++ lib.optional (pythonOlder "3.7") dataclasses
+    ++ lib.optional (pythonOlder "3.8") typing-extensions;
 
   meta = with lib; {
     description = "The uncompromising Python code formatter";
-    homepage    = "https://github.com/psf/black";
-    changelog   = "https://github.com/psf/black/blob/${version}/CHANGES.md";
-    license     = licenses.mit;
+    homepage = "https://github.com/psf/black";
+    changelog = "https://github.com/psf/black/blob/${version}/CHANGES.md";
+    license = licenses.mit;
     maintainers = with maintainers; [ sveitser ];
   };
 }

